wolfcrypt/src/asn.c

```diff
@@ -20631,7 +20636,7 @@ static int DecodeBasicCaConstraint(const byte* input, int sz, DecodedCert* cert)
     if (ret == 0) {
         /* Get the CA boolean and path length when present. */
         GetASN_Boolean(&dataASN[BASICCONSASN_IDX_CA], &isCA);
-        GetASN_Int8Bit(&dataASN[BASICCONSASN_IDX_PLEN], &cert->pathLength);
+        GetASN_Int16Bit(&dataASN[BASICCONSASN_IDX_PLEN], &cert->pathLength);
```

Comment:
Why aren't we testing WOLFSSL_MAX_PATH_LEN here? It might be useful to add comment on these indicating that WOLFSSL_MAX_PATH_LEN can be overridden if someone needs more than 127. The reason for the limit has to do with memory usage.

Reply:
WOLFSSL_MAX_PATH_LEN is tested in the ASN template case below on line 20656.  I can add a comment, now that this is always a 16-bit variable we are slightly increasing memory usage.
I can look into making this dynamic where we only use a 16-bit integer if WOLFSSL_MAX_PATH_LEN > 127.